Add instruction for setup (#58)
OneSock-inc/OnlyOne: backend/README.md

API is documented in the [notion](https://www.notion.so/webblitchy/Sp-cification-API-049c86e231324f048c2d2569b49a30ac). You might want to download [Postman](https://www.postman.com/) to make api calls.

# Developper information

## Project's description 
In this folder you can see the backend of the onlyOne project.
The project is a reponsive website where someone can signup, login and propose lonely sock to match. We try to match two lonely sock together to form a new happy pair. 
In this folder you can see the backend server code. it's written in Go/Golang using the [gin](https://gin-gonic.com/) framework.

## Structure of the backend's code

The backend is divided in two parts (go modules):
 1. API
 2. DB  

The api module is composed of the code setting up the routes and the handlers for the requests/response. The JWT middleware used for the authentification is also defined there.

Most of these handlers just do a call to the DB module where the heavy lifting (db transaction, commit rollback in case of error) is done. Then the handlers return the response to the client or the error .
The databases is totaly abstracted from the API viewpoint

> The DB module is where all the grpc calls are made to the database. We don't do grpc call ourself we only use the grpc client offered by the firestore lib. The matching algorithmes is implemented there. We also take care of the error handling there, for exemple if the data send by a user to a route / response handler is ill formed or simply missing we detect it there before creating a db record . 

Reviewer: ```suggestion
The DB module is where all the [gRPC](https://grpc.io/) calls are made to the database. We don't do gRPC call ourself as we only use the gRPC client offered by the firestore library. The matching algorithms is implemented there. We also take care of the error handling there. For example, if the data sent by a user to a route / response handler is ill formed or simply missing, we detect it there before creating a database record . 
```

> The DB module is where all the [grpc](https://grpc.io) calls are made to the database. We don't do grpc call ourself, inestead we use the firestore library that do the grpcs call itself. The matching algorithme is implemented there. We also take care of the error handling there, for exemple if the data send by a user to a route / response handler is ill formed or simply missing we detect it there before creating saving it to the database. 

Reviewer: ```suggestion
The DB module is where all the [grpc](https://grpc.io) calls are made to the database. We don't call any grpc ourself, instead we use the firestore library (which calls the grpcs). The matching algorithm is implemented there. We also take care of the error handling there, for example, if the data send by a user to a route / response handler is ill formed or simply missing we detect it there before creating saving it to the database. 
```
- The sentence `The matching algorithm is implemented there` is not obvious: where ?
- Why do we speak about grpc ?

> The DB module is where all the calls to the database are made. We use the firestore [library](https://pkg.go.dev/google.golang.org/cloud/firestore) . The matching algorithm is implemented in this module currently this is a [knn](https://en.wikipedia.org/wiki/K-nearest_neighbors_algorithm) search in a [kdtree](https://github.com/sjwhitworth/golearn/blob/master/kdtree/kdtree.go). We also take care of the error handling related to the db there, for example if the data send by a user to a route / response handler is ill formed or simply missing we detect it there before saving it to the database. 

Reviewer: ```suggestion
The DB module is where all the calls to the database are made. We use the firestore [library](https://pkg.go.dev/google.golang.org/cloud/firestore). The matching algorithm is also implemented in this module, and it is currently a [knn](https://en.wikipedia.org/wiki/K-nearest_neighbors_algorithm) search in a [kdtree](https://github.com/sjwhitworth/golearn/blob/master/kdtree/kdtree.go). We also take care of the error handling related to the database there, for example if the data sent by a user to a route / response handler is ill formed or simply missing, we detect it there before saving it to the database. 
```
